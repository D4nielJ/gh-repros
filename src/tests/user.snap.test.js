import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';

import { configureStore } from '@reduxjs/toolkit';
import { combineReducers } from 'redux';

// Import your own reducer
import userReducer from '../features/user/userSlice'
import User from '../features/user/user';

jest.mock('../features/user/userSlice');

let preloadedState = {};

let reducer = combineReducers({
  user: userReducer,
});

const store = configureStore({ reducer, preloadedState });

it('renders correctly the user', () => {
  const userTree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <User id={'d4nielj'} />
        </Router>
      </Provider>,
    )
    .toJSON();

  expect(userTree).toMatchSnapshot();
});
