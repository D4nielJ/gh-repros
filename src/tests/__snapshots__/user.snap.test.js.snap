// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly the user 1`] = `
<section
  className="min-h-screen bg-bh-lightBlue"
>
  <div
    className="bg-bh-darkBlue flex justify-between items-center px-4 py-2"
  >
    <button
      className="text-white cursor-pointer hover:text-gray-200 transition-all"
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="octicon octicon-chevron-left"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path fill-rule=\\"evenodd\\" d=\\"M9.78 12.78a.75.75 0 01-1.06 0L4.47 8.53a.75.75 0 010-1.06l4.25-4.25a.75.75 0 011.06 1.06L6.06 8l3.72 3.72a.75.75 0 010 1.06z\\"></path>",
          }
        }
        fill="currentColor"
        height={20}
        role="img"
        style={
          Object {
            "display": "inline-block",
            "overflow": "visible",
            "userSelect": "none",
            "verticalAlign": "text-bottom",
          }
        }
        viewBox="0 0 16 16"
        width={20}
      />
    </button>
    <span
      className="uppercase text-white text-sm font-bold"
    >
      D4niel J
    </span>
    <button
      className="text-white cursor-pointer hover:text-gray-200 transition-all"
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="octicon octicon-search"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path fill-rule=\\"evenodd\\" d=\\"M11.5 7a4.499 4.499 0 11-8.998 0A4.499 4.499 0 0111.5 7zm-.82 4.74a6 6 0 111.06-1.06l3.04 3.04a.75.75 0 11-1.06 1.06l-3.04-3.04z\\"></path>",
          }
        }
        fill="currentColor"
        height={20}
        role="img"
        style={
          Object {
            "display": "inline-block",
            "overflow": "visible",
            "userSelect": "none",
            "verticalAlign": "text-bottom",
          }
        }
        viewBox="0 0 16 16"
        width={20}
      />
    </button>
  </div>
  <div
    className="flex"
  >
    <div
      className="w-44 py-4 pl-6"
    >
      <div
        className="w-36 h-36"
      >
        <div
          className="aspect-h-1 aspect-w-1 relative rounded-full border-2 border-bh-darkBlue"
        >
          <img
            alt="d4nielj"
            className="bg-bh-blue inset-0 w-full h-full object-cover absolute rounded-full"
            src="https://avatars.githubusercontent.com/u/79114233?v=4"
          />
        </div>
      </div>
    </div>
    <div
      className="w-1/2 px-3 flex flex-col justify-center"
    >
      <h2
        className="font-bold text-white text-2xl leading-tight"
      >
        D4niel J
      </h2>
      <a
        className="text-white underline mb-2 text-sm hover:text-gray-200 transition-all"
        href="https://github.com/D4nielJ"
        rel="noreferrer"
        target="_blank"
      >
        <h3>
          @
          D4nielJ
        </h3>
      </a>
      <div
        className="text-sm text-white"
      >
        <div>
          <svg
            aria-hidden="true"
            className="mr-1"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<path fill-rule=\\"evenodd\\" d=\\"M10.5 5a2.5 2.5 0 11-5 0 2.5 2.5 0 015 0zm.061 3.073a4 4 0 10-5.123 0 6.004 6.004 0 00-3.431 5.142.75.75 0 001.498.07 4.5 4.5 0 018.99 0 .75.75 0 101.498-.07 6.005 6.005 0 00-3.432-5.142z\\"></path>",
              }
            }
            fill="currentColor"
            height={16}
            role="img"
            style={
              Object {
                "display": "inline-block",
                "overflow": "visible",
                "userSelect": "none",
                "verticalAlign": "text-bottom",
              }
            }
            viewBox="0 0 16 16"
            width={16}
          />
          <span
            className="mr-1"
          >
            <span
              className="font-bold"
            >
              28
            </span>
             
            Following
          </span>
        </div>
        <div>
          <svg
            aria-hidden="true"
            className="mr-1"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<path fill-rule=\\"evenodd\\" d=\\"M5.5 3.5a2 2 0 100 4 2 2 0 000-4zM2 5.5a3.5 3.5 0 115.898 2.549 5.507 5.507 0 013.034 4.084.75.75 0 11-1.482.235 4.001 4.001 0 00-7.9 0 .75.75 0 01-1.482-.236A5.507 5.507 0 013.102 8.05 3.49 3.49 0 012 5.5zM11 4a.75.75 0 100 1.5 1.5 1.5 0 01.666 2.844.75.75 0 00-.416.672v.352a.75.75 0 00.574.73c1.2.289 2.162 1.2 2.522 2.372a.75.75 0 101.434-.44 5.01 5.01 0 00-2.56-3.012A3 3 0 0011 4z\\"></path>",
              }
            }
            fill="currentColor"
            height={16}
            role="img"
            style={
              Object {
                "display": "inline-block",
                "overflow": "visible",
                "userSelect": "none",
                "verticalAlign": "text-bottom",
              }
            }
            viewBox="0 0 16 16"
            width={16}
          />
          <span
            className="mr-1"
          >
            <span
              className="font-bold"
            >
              22
            </span>
             
            Followers
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="px-4 py-1 text-white  bg-bh-blue flex justify-between items-center"
  >
    <div
      className="flex"
    >
      <span
        className="mr-1"
      >
        <svg
          aria-hidden="true"
          className="mr-1"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<path fill-rule=\\"evenodd\\" d=\\"M2 2.5A2.5 2.5 0 014.5 0h8.75a.75.75 0 01.75.75v12.5a.75.75 0 01-.75.75h-2.5a.75.75 0 110-1.5h1.75v-2h-8a1 1 0 00-.714 1.7.75.75 0 01-1.072 1.05A2.495 2.495 0 012 11.5v-9zm10.5-1V9h-8c-.356 0-.694.074-1 .208V2.5a1 1 0 011-1h8zM5 12.25v3.25a.25.25 0 00.4.2l1.45-1.087a.25.25 0 01.3 0L8.6 15.7a.25.25 0 00.4-.2v-3.25a.25.25 0 00-.25-.25h-3.5a.25.25 0 00-.25.25z\\"></path>",
            }
          }
          fill="currentColor"
          height={16}
          role="img"
          style={
            Object {
              "display": "inline-block",
              "overflow": "visible",
              "userSelect": "none",
              "verticalAlign": "text-bottom",
            }
          }
          viewBox="0 0 16 16"
          width={16}
        />
        <span
          className="font-bold"
        >
          32
        </span>
      </span>
      <h3
        className="uppercase"
      >
        Repos
      </h3>
    </div>
    <div
      className="flex"
    >
      <p
        className="mr-1"
      >
        Sort by:
      </p>
      <select
        className="text-bh-black"
        onChange={[Function]}
        value="Stars"
      >
        <option
          value="Stars"
        >
          Stars
        </option>
        <option
          value="Forks"
        >
          Forks
        </option>
        <option
          value="Size"
        >
          Size
        </option>
      </select>
    </div>
  </div>
  <p
    className="text-center pt-4 text-white font-bold"
  >
    No repos available :c, try sorting the list
  </p>
</section>
`;
